test/happy flow.ts

```diff
@@ -0,0 +1,72 @@
+import { Builder, By, until, WebDriver } from "selenium-webdriver";
+import { describe, it, beforeEach, afterEach } from "mocha";
+import assert from "assert";
+
+let driver: WebDriver;
+
+describe('SauceDemo Happy Flow Test', function () {
+  this.timeout(30000);
+
+  beforeEach(async () => {
+    driver = await new Builder().forBrowser('chrome').build();
+  });
+
+  afterEach(async () => {
+    await driver.quit();
+  });
+
+  it('should login, add items to cart, verify cart, and complete checkout', async () => {
+    await driver.get('https://www.saucedemo.com/');
+    await driver.findElement(By.name('user-name')).sendKeys('standard_user');
+    await driver.findElement(By.id('password')).sendKeys('secret_sauce');
+    await driver.findElement(By.id('login-button')).click();
+
+    await driver.wait(until.urlIs('https://www.saucedemo.com/inventory.html'), 5000);
```

Comment:
check the URL - very ad and unreliable assertion, replace it with any other: e.g. verify that element is present on the current screen